File: Olympiad/COCI/official/2007/contest1_solutions/modulo.c

/*
  Croatian Open Competition in Informatics 2006/2007
  Contest 1 - Task MODULO
  Programming language C
*/

#include <stdio.h>

int main() {
   int i, sol = 0;
   int mods[42] = {0};

   for ( i=0; i<10; ++i ) {
      int x;
      scanf( "%d", &x );
      mods[ x%42 ] = 1;
   }
   for ( i=0; i<42; ++i )
      sol += mods[i];

   printf( "%d\n", sol );

   return 0;
}

File: Olympiad/COCI/official/2007/contest1_solutions/modulo.pas

(*
  Croatian Open Competition in Informatics 2006/2007
  Contest 1 - Task MODULO
  Programming language Pascal
*)

program modulo;

var
   i, sol, x : integer;
   pojavio   : array[0..41] of boolean;

begin
   for i:=1 to 10 do begin
      read(x);
      pojavio[x mod 42] := true;
   end;

   for i:=0 to 41 do
      if pojavio[i] then
         sol := sol + 1;

   writeln( sol );
end.
